src/utils/helpers.js

```diff
@@ -12,27 +13,97 @@ export function getClosestParksByAmenityTypeAndID(
     (park) =>
       park[type].includes(id) &&
       currentLocation &&
-      haversine(currentLocation, {
-        latitude: park.Latitude,
-        longitude: park.Longitude,
-      }) <= distanceFilter
+      haversine(currentLocation, park.location) <= defaultDistanceFilter
   )
 
   if (currentLocation) {
     parksByType.sort((a, b) => {
-      const distanceToA = haversine(currentLocation, {
-        latitude: a.Latitude,
-        longitude: a.Longitude,
-      })
-
-      const distanceToB = haversine(currentLocation, {
-        latitude: b.Latitude,
-        longitude: b.Longitude,
-      })
+      const distanceToA = haversine(currentLocation, a.location)
+      const distanceToB = haversine(currentLocation, b.location)
 
       return distanceToA - distanceToB
     })
   }
 
   return parksByType
 }
+
+export function filterParks({
+  parks,
+  searchTerm,
+  location,
+  distance,
+  activities,
+  facilities,
+}) {
+  /*
+   * Filter parks based on search string
+   */
+  const searchedParks =
+    searchTerm !== undefined
+      ? parks.filter((park) =>
+          park.searchableTitle.toLowerCase().includes(searchTerm.toLowerCase())
+        )
+      : parks
+
+  /*
+   * Further filter parks based on distance, activity, and facility filters
+   */
+  const filterDistance = distance >= 99 ? maxDistance : distance
```

Comment:
`Math.min()` is a little more readable to me for calculations like this.
```suggestion
  const filterDistance = Math.min(distance, maxDistance)
```

Reply:
I understand why `Math.min()` is your preferred approach for readable calculation. I agree however that is not what I'm trying to achieve here. I want to use the provided filter distance until that distance reaches 100km at which point we basically want to remover the filter distance or "max it out" which is what I'm doing here. After 100km the distance filter would return all parks. Does that make sense?

Reply:
Oh sorry, I was going off memory and thought `maxDistance` was `100`. I guess I mixed that up with the `defaultDistanceFilter`.

LGTM 👍